plugins/JsTrackerInstallCheck/API.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\Plugins\JsTrackerInstallCheck;
+
+use Piwik\Piwik;
+
+/**
+ * @internal
+ */
+class API extends \Piwik\Plugin\API
+{
+    /**
+     * @var JsTrackerInstallCheck
+     */
+    protected $jsTrackerInstallCheck;
+
+    public function __construct(JsTrackerInstallCheck $jsTrackerInstallCheck)
+    {
+        $this->jsTrackerInstallCheck = $jsTrackerInstallCheck;
+    }
+
+    /**
+     * Check whether a test request has been recorded for the provided nonce
+     *
+     * @param int $idSite
+     * @param string $nonce
+     * @return array list of containers ['isSuccess' => true]
+     * @throws \Exception If the user doesn't have the right permissions
+     */
+    public function checkForJsTrackerInstallTestSuccess(int $idSite, string $nonce): array
+    {
+        Piwik::checkUserHasViewAccess($idSite);
+
+        return ['isSuccess' => $this->jsTrackerInstallCheck->checkForJsTrackerInstallTestSuccess($idSite, $nonce)];
+    }
+
+    /**
+     * Check whether a test request has been recorded for the provided site
+     *
+     * @param int $idSite
+     * @return array list of containers ['isSuccess' => true]
+     * @throws \Exception If the user doesn't have the right permissions
+     */
+    public function getJsTrackerInstallTestResult(int $idSite): array
+    {
+        Piwik::checkUserHasViewAccess($idSite);
+
+        return ['isSuccess' => $this->jsTrackerInstallCheck->checkForJsTrackerInstallTestSuccess($idSite)];
+    }
```

Comment:
I was trying to prevent an accidental false positive if for some reason the nonce didn't get passed while trying to check a specific nonce, but I can combine the methods.